src/state.rs

```diff
@@ -174,15 +174,18 @@ impl<ID: TreeId, TM: TreeMeta, A: Actor> State<ID, TM, A> {
             match op1.timestamp().cmp(&self.log_op_list[0].timestamp()) {
                 Ordering::Equal => {
                     // This case should never happen in normal operation
-                    // because it is required that all timestamps are unique.
+                    // because it is requirement/invariant that all
+                    // timestamps are unique.  However, uniqueness is not
+                    // strictly enforced in this impl.
                     // The crdt paper does not even check for this case.
                     //
-                    // We can throw an exception to catch it during dev/test.
-                    // #[cfg(debug_assertions)]
-                    // panic!("applying op with timestamp equal to previous op.  Every op should have a unique timestamp.");
+                    // We can panic to catch it during dev/test.
+                    #[cfg(debug_assertions)]
+                    panic!("applying op with timestamp equal to previous op.  Every op must have a unique timestamp.");
 
-                    // Production code should just treat it as a non-op.
-                    // #[cfg(not(debug_assertions))]
+                    // Production code can just treat it as a no-op.
+                    #[cfg(not(debug_assertions))]
+                    return;
```

Comment:
I feel here we should do one thing and do it consistently. So if it is an error we need to at least log it or return an error. The latter is problematic though as rust-crdt trait requires no return type (no Result). So here I feel log is our best bet in both debug and release.

Reply:
yes, I think that's good.  will make it so.